core/arch/arm/mm/core_mmu.c

```diff
@@ -222,6 +226,99 @@ static struct tee_mmap_region *find_map_by_pa(unsigned long pa)
 	return NULL;
 }
 
+extern const struct core_mmu_phys_mem __start_phys_sdp_mem_section;
+extern const struct core_mmu_phys_mem __end_phys_sdp_mem_section;
+
+static bool pbuf_is_sdp_mem(paddr_t pbuf, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+
+	for (mem = &__start_phys_sdp_mem_section;
+	     mem < &__end_phys_sdp_mem_section; mem++)
+		if (core_is_buffer_inside(pbuf, len, mem->addr, mem->size))
+			return true;
+
+	return false;
+}
+
+#define MSG_SDP_INSTERSECT(pa1, sz1, pa2, sz2) \
+	EMSG("[%" PRIxPA " %" PRIxPA "] intersecs [%" PRIxPA " %" PRIxPA "]", \
+			pa1, pa1 + sz1, pa2, pa2 + sz2)
+
+/* check SDP memories comply with registered memories */
+static void verify_sdp_mem_areas(struct tee_mmap_region *mem_map, size_t len)
+{
+	const struct core_mmu_phys_mem *mem;
+	const struct core_mmu_phys_mem *mem2;
+	const struct core_mmu_phys_mem *start = &__start_phys_sdp_mem_section;
+	const struct core_mmu_phys_mem *end = &__end_phys_sdp_mem_section;
+	struct tee_mmap_region *mmap;
+	size_t n;
+
+#ifndef CFG_SECURE_DATA_PATH
+	if (start != end)
+		panic("SDP memory found but CFG_SECURE_DATA_PATH is disabled");
+#endif
+
+	for (mem = start; mem < end; mem++)
```

Comment:
It would be nice to not have any of this code (or possibly have it optimized away by the compiler) when `CFG_SECURE_DATA_PATH` isn't defined.